test/evp_test.c

```diff
@@ -1997,7 +1997,8 @@ static int keygen_test_run(EVP_TEST *t)
     t->err = NULL;
     if (EVP_PKEY_keygen(keygen->genctx, &pkey) <= 0) {
         t->err = "KEYGEN_GENERATE_ERROR";
-        goto err;
+        EVP_PKEY_free(pkey);
```

Comment:
Wait what???  This seems wrong, it's absurd to expect that a failing EVP_PKEY_keygen() produces an EVP_PKEY.  I'd say this is a bug in 3.0!

Reply:
Oh...this might not actually be necessary

Reply:
The important point was the change in return value (0 -> 1)

Reply:
0 means "I wasn't even able to run the test", 1 means "I ran the test successfully (but the thing I tried to do might have failed - and that could be ok and expected)"